site/js/ledAPI.js

```diff
@@ -0,0 +1,256 @@
+
+            /* update currentDeviceLabel shown to user */
+            $('#currentDeviceLabel').html(this.value);
+            document.getElementById('currentDeviceLabel').value =
+                this.value;
+            console.info("deviceID set to: ", deviceID);
+            $('#currentDeviceLabel').show();
+
+
+            /* If selected device is 'new', get device settings from server and update page */
+            if (lastDevice != deviceID) {
+                console.debug("lastDevice != deviceID ---> " +
+                    lastDevice + " != " + deviceID);
+
+                $apiUtils.getData(apiUrl, deviceID,
+                    updateMainContent, disableMainContent);
+                $('#brightness').toggleClass('look-disabled', false);
+
+                if (lastDevice && document.getElementById(
+                        lastDevice).classList.contains(
+                        'btn-current')) {
+                    document.getElementById(lastDevice).classList.remove(
+                        'btn-current');
+                }
+            }
+
+            console.debug("setting device button to current");
+            document.getElementById(deviceID).classList.add(
+                'btn-current');
+        });
+
+
+        /* Change name of device when form is submitted */
+        $('form').on('submit', function changeDeviceName(event) {
+
+            /* Prevent the page from reloading after form submit */
+            event.preventDefault();
+
+            /* Set the text-output span to the value of the first input */
+            var $input = $(this).find('input');
+            var input = $input.val();
+
+            /* update device name on server */
+            $apiUtils.putRequest(apiUrl, deviceID, 'name', input);
+
+            /*  update displayed name for device button */
+            $('#currentDeviceLabel').html(input.slice(0,
+                nameMaxLength));
+            document.getElementById(deviceID).value = input.slice(0,
+                nameMaxLength);
+
+        });
+
+    }
+
+    function updateOnButton(device) {
+        console.debug("updateOnButton: state received: ", device[1][
+            'onState'
+        ]);
+        switch (device[1]['onState']) {
+            case "true":
+                $("#onSwitch").prop("checked", true);
+                break;
+            case "false":
+                $("#onSwitch").prop("checked", false);
+                break;
+        }
+    }
+
+
+    function updateBrightSlider(device) {
+        var brightVal = device[1]['brightness']
+        $('#brightness-value').html(brightVal);
+        $('#slider-brightness').val(brightVal);
+    }
+
+    function updateDeviceList(newDevices) {
+        var remove = [];
+        var add = [];
+
+        newDevices.sort();
+
+        /* Identify old devices no longer online and remove */
+        for (var i in devices) {
+            if (newDevices.indexOf(devices[i]) === -1) remove.push(devices[
```

Comment:
I think this is probably fine as is, but some pedantic interviewers might point out that this becomes somewhat inefficient if you have huge arrays in devices and newDevices. If this comes up, you could pre-compute a `Set` of `newDevices` which would allow you to check if a device was included more quickly using `has`...

eg:
```javascript
// faster for large sets of devices + newDevices.
// for a list of length n, building the set takes O(n*log(n)) time,
// Set.prototype.has() takes O(log(n)) time
const newDeviceSet = new Set(newDevices);
for (const device of devices)
    if (!newDeviceSet.has(device)
        remove.push(device);
```